python/version_details/version_details.py

```diff
@@ -0,0 +1,1213 @@
+            self.clear()
+            return
+
+        # Switch over to the page that contains the primary display
+        # widget set now that we have data to show.
+        self.ui.pages.setCurrentWidget(self.ui.main_page)
+
+        # If there aren't any fields set in the info widget then it
+        # likely means we're loading from a "cleared" slate and need
+        # to re-add our relevant fields.
+        if not self.ui.current_version_card.fields:
+            self.ui.current_version_card.fields = self._active_fields
+            self.ui.current_version_card.label_exempt_fields = self._persistent_fields
+
+        self.ui.note_stream_widget.load_data(entity)
+
+        shot_filters = [["id", "is", entity["id"]]]
+        self.version_info_model.load_data(
+            entity_type="Version",
+            filters=shot_filters,
+            fields=self._fields,
+        )
+
+        for note_id in self.note_threads.keys():
+            self._process_note_arrival(note_id)
+
+        self.entity_loaded.emit(entity)
+
+    def save_preferences(self):
+        """
+        Saves user preferences to disk.
+        """
+        self._settings_manager.store(
+            VersionDetailsWidget.FIELDS_PREFS_KEY,
+            self._fields,
+            self._settings_manager.SCOPE_ENGINE,
+        )
+
+        self._settings_manager.store(
+            VersionDetailsWidget.ACTIVE_FIELDS_PREFS_KEY,
+            self._active_fields,
+            self._settings_manager.SCOPE_ENGINE,
+        )
+
+        self._settings_manager.store(
+            VersionDetailsWidget.VERSION_LIST_FIELDS_PREFS_KEY,
+            self.version_delegate.fields,
+            self._settings_manager.SCOPE_ENGINE,
+        )
+
+    def set_note_metadata(self, note_id, metadata):
+        """
+        Sets a Note entity's metadata in Shotgun.
+
+        :param int note_id: The Note entity id.
+        :param str metadata: The metadata to set in Shotgun.
+        """
+        self._note_set_metadata_uids.append(
+            self._data_retriever.execute_update(
+                "Note",
+                note_id,
+                {self.NOTE_METADATA_FIELD:metadata},
+            )
+        )
+
+    def set_note_screenshot(self, image_path):
+        """
+        Takes the given file path to an image and sets the new note
+        widget's thumbnail image.
+
+        :param str image_path:  A file path to an image file on disk.
+        """
+        self.ui.note_stream_widget.note_widget._set_screenshot_pixmap(
+            QtGui.QPixmap(image_path),
+        )
+
+    def setParent(self, parent):
+        """
+        Calls the base class' method of the same name, and then checks
+        to see if the parent is a dock widget. If it is, then a custom
+        title bar is made visible.
+        """
+        super(VersionDetailsWidget, self).setParent(parent)
+
+        try:
+            self.parent().dockLocationChanged.connect(self._dock_location_changed)
```

Comment:
Likely not. It's a bit of a hack to get the right look in RV without breaking when used in other host applications.